pgml-apps/pgml-chat/pgml_chat/pgml_playground.py

```diff
@@ -0,0 +1,114 @@
+from pgml import Collection, Builtins, Pipeline
+import asyncio
+from rich import print
+
+    # chat_history_user_messages = await chat_collection.query().vector_recall(user_input, chat_history_pipeline, query_params)(
+    #     {   
+    #         "limit" : chat_history,
+    #         "filter": {
+    #             "metadata": {
+    #                 "$and": [
+    #                     {{"role": {"$eq": "user"}},
+    #                     {{"interface": {"$eq": chat_interface}},
+    #                 ]
+    #             }
+    #         },
+    #     }
+    # ).fetch_all()
+
+    # chat_history_assistant_messages = await chat_collection.query().vector_recall(user_input, chat_history_pipeline, query_params)(
+    #     {   
+    #         "limit" : chat_history,
+    #         "filter": {
+    #             "metadata": {
+    #                 "$and": [
+    #                     {"role": {"$eq": "assistant"}},
+    #                     {"interface": {"$eq": chat_interface}},
+    #                 ]
+    #             }
+    #         },
+    #     }
+    # ).fetch_all()
+
+    
+async def main():
+    collection = Collection("pgml_chat_all_docs_4_chat_history")
+    builtins = Builtins()
+    query = """SELECT metadata->>'role' as role, text as content from %s.documents
+            WHERE metadata @> '{\"interface\" : \"cli\"}'::JSONB 
+            AND metadata @> '{\"role\" : \"user\"}'::JSONB 
+            OR metadata @> '{\"role\" : \"assistant\"}'::JSONB 
+            ORDER BY metadata->>'timestamp' DESC LIMIT %d""" % (
```

Comment:
Why is the timestamp in the metadata, instead of a field on the record?

Reply:
We have a field on the record. I insert user message and the response together and each of them will have different timestamps. I could also order by the field on the record.